aws/src/main/java/org/apache/iceberg/aws/LakeFormationAwsClientFactory.java

```diff
@@ -0,0 +1,230 @@
+
+    private AwsSessionCredentials getAssumeRoleCredentials(Map<String, String> properties) {
+      String roleArn = properties.get(AwsProperties.CLIENT_ASSUME_ROLE_ARN);
+      Preconditions.checkNotNull(roleArn, "Cannot initialize LakeFormationAwsClientFactory with empty roleArn");
+      String sessionName = String.format("iceberg-aws-%s", UUID.randomUUID());
+      String externalId = properties.get(AwsProperties.CLIENT_ASSUME_ROLE_EXTERNAL_ID);
+      String tagString = properties.get(AwsProperties.CLIENT_ASSUME_ROLE_TAGS);
+      Preconditions.checkNotNull(tagString, "Cannot initialize LakeFormationAwsClientFactory with empty tags");
+      List<String> tokens = Splitter.on(',').splitToList(tagString);
+      String key = tokens.get(0).substring("Key=".length());
+      String value = tokens.get(1).substring("Value=".length());
+      Tag tag = Tag.builder().key(key).value(value).build();
+      int timeout = PropertyUtil.propertyAsInt(properties,
+          AwsProperties.CLIENT_ASSUME_ROLE_TIMEOUT_SEC, AwsProperties.CLIENT_ASSUME_ROLE_TIMEOUT_SEC_DEFAULT);
+      AssumeRoleRequest request = AssumeRoleRequest.builder()
+          .roleArn(roleArn)
+          .roleSessionName(sessionName)
+          .durationSeconds(timeout)
+          .externalId(externalId)
+          .tags(tag)
+          .build();
+      return (AwsSessionCredentials) StsAssumeRoleCredentialsProvider.builder()
+          .stsClient(StsClient.builder().httpClientBuilder(UrlConnectionHttpClient.builder()).build())
+          .refreshRequest(request)
+          .build().resolveCredentials();
+    }
+
+    private LakeFormationClient buildLakeFormationClient(String accessKey,
+        String secretKey, String sessionToken, String region) {
+      StaticCredentialsProvider credentialsProvider =  StaticCredentialsProvider.create(AwsSessionCredentials.create(
+          accessKey, secretKey, sessionToken));
+      return LakeFormationClient.builder().credentialsProvider(credentialsProvider).region(Region.of(region)).build();
+    }
+
+    private String getTableArn(Map<String, String> properties) {
+      String dbName = properties.get(AwsProperties.LAKE_FORMATION_DB_NAME_KEY);
+      String tableName = properties.get(AwsProperties.LAKE_FORMATION_TABLE_NAME_KEY);
+      String accountId = properties.get(AwsProperties.LAKE_FORMATION_ACCOUNT_ID);
+      String region = properties.get(AwsProperties.LAKE_FORMATION_AWS_REGION);
+
+      String relativeId = String.format("table/%s/%s", dbName, tableName);
+      String partitionName = PartitionMetadata.of(Region.of(region)).name();
+      return String.format("arn:%s:%s:%s:%s:%s",
+          partitionName,
+          GLUE_SERVICE_NAME,
+          region,
+          accountId,
+          relativeId);
+    }
+
+    private List<String> getPrivilegeList(Map<String, String> properties) {
+      // TODO: load PrivilegeList from properties dynamically for different operations
+      return Arrays.asList("SELECT", "INSERT", "DELETE");
```

Comment:
this can be moved as a private static final variable for now. For Iceberg we never know what is the caller's intention, so we cannot really narrow down the list, at least in the short term.